We have several configuration options for "analytic events", which are called transaction events. The term analytic events is an older term. We've updated the language for these configuration options to reference transaction events instead, and deprecated the older analytic event configs.  Note: While I did update all references to the deprecated configuration options in the main agent code, I did not rename other references to analytic events in the internal agent code (such as method/variable names).

tannalynn opened the pull request:
# Overview

We have several configuration options for "analytic events", which are called transaction events. The term analytic events is an older term. We've updated the language for these configuration options to reference transaction events instead, and deprecated the older analytic event configs.

Note: While I did update all references to the deprecated configuration options in the main agent code, I did not rename other references to analytic events in the internal agent code (such as method/variable names).


# Related Github Issue
Include a link to the related GitHub issue, if applicable

# Testing
The agent includes a suite of unit and functional tests which should be used to
verify your changes don't break existing functionality. These tests will run with 
Github Actions when a pull request is made. More details on running the tests locally can be found 
[here for our unit tests](https://github.com/newrelic/newrelic-ruby-agent/blob/main/test/README.md), 
and [here for our functional tests](https://github.com/newrelic/newrelic-ruby-agent/blob/main/test/multiverse/README.md).
For most contributions it is strongly recommended to add additional tests which
exercise your changes. 

# Reviewer Checklist
- [ ] Perform code review
- [ ] Add performance label
- [ ] Perform appropriate level of performance testing
- [ ] Confirm all checks passed
- [ ] Add version label prior to acceptance

kaylareopelle reviewed (approved):
This looks good to me, Tanna. The only code-based change that comes to mind is to add a positive test case to verify config callbacks will work with procs in the future.

It may also be worth adding a sentence to the deprecation issue we discussed on the call earlier today with a line item to replace references to analytics events with transaction events inside the code to prevent future confusion about what analytics events are.

Thanks for your innovation! 🧠

tannalynn commented:
> This looks good to me, Tanna. The only code-based change that comes to mind is to add a positive test case to verify config callbacks will work with procs in the future.

Added this test! Thanks!

> It may also be worth adding a sentence to the deprecation issue we discussed on the call earlier today with a line item to replace references to analytics events with transaction events inside the code to prevent future confusion about what analytics events are.

I added the new issue for removing deprecated configs. I'm not sure we can remove all of the references to analytic events in the internal agent code because that's still what the endpoint is called and other things related to the server.

kaylareopelle commented:
>I added the new issue for removing deprecated configs. I'm not sure we can remove all of the references to analytic events in the internal agent code because that's still what the endpoint is called and other things related to the server.

@tannalynn  - I didn't realize analytic events was still referenced in the overall structure of the system! No need to remove it then. :)

kaylareopelle commented on CHANGELOG.md:
```diff
@@ -2,6 +2,14 @@
   ## v8.5.0
 
+  * **Added updated configuration options for transaction events and deprecate previous configs**
+    This release deprecates and replaces the following configuration options
```
```suggestion
    This release deprecates and replaces the following configuration options:
```

kaylareopelle replied:
That's ok! No need to go back and change it now. We usually review the Changelog for little things like that before a release anyway. 😄

kaylareopelle reviewed (approved):
🌟
